src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4090,9 +4115,12 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	List *groupClauseList = originalOpNode->groupClauseList;
 	List *sortClauseList = originalOpNode->sortClauseList;
 	List *targetList = originalOpNode->targetList;
+	bool orderByNonAggregates = !(HasOrderByAggregate(sortClauseList, targetList));
```

Comment:
HasOrderByAggregate() checks if we have an order by an aggregate function. The opposite is not necessarily an order by without aggregates. There could be no order by at all. 

I checked other usages and the place you copy this. It always sits inside a if (sortClauseList != NIL) block, so we know that there is an order by clause at least.